src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatastore.scala

```diff
@@ -0,0 +1,231 @@
+package tech.cryptonomic.conseil.tezos
+
+import tech.cryptonomic.conseil.tezos.Tables.{BalanceUpdatesRow, OperationGroupsRow, OperationsRow}
+import tech.cryptonomic.conseil.tezos.TezosTypes.{
+  Account,
+  AccountId,
+  Block,
+  BlockReference,
+  BlockTagged,
+  Delegate,
+  PublicKeyHash,
+  Voting
+}
+import tech.cryptonomic.conseil.tezos.repositories.{
+  AccountsRepository,
+  BlocksRepository,
+  DelegatesRepository,
+  OperationsRepository,
+  VotingRepository
+}
+import tech.cryptonomic.conseil.util.Conversion
+
+import cats.implicits._
+import cats.effect._
+import cats.data.Kleisli
+import slick.dbio.DBIO
+import slick.jdbc.PostgresProfile.api._
+import slickeffect.implicits._
+import scala.concurrent.ExecutionContext
+import com.typesafe.scalalogging.LazyLogging
+
+/**
+  * Functions for writing Tezos data to a database.
+  */
+class TezosDatastore(
```

Comment:
Naively speaking, a data store would make sense when there are several sources / sinks of data. At the moment, however, we simply use a database with tables. What is the value of the data store and repository abstraction?

Reply:
Maybe it's a naming problem... I simply needed a more generic term that would compose complex persistence operations, i.e. multiple, compound queries or queries against different types of entities.

The Repository abstraction is concerned about a single entity, the "datastore" exposes more complex operations to the application.